lib/maths/CDataFrameCategoryEncoder.cc

```diff
@@ -271,15 +271,9 @@ std::uint64_t CDataFrameCategoryEncoder::checksum(std::uint64_t seed) const {
 }
 
 void CDataFrameCategoryEncoder::acceptPersistInserter(core::CStatePersistInserter& inserter) const {
-    auto persistEncodingVector = [this](core::CStatePersistInserter& inserter) {
-        for (const auto& encoding : m_Encodings) {
-            auto acceptPersistInserter = [&encoding](core::CStatePersistInserter& inserter) {
```

Comment:
The nested parameter names were generating compiler warnings for me. I decided to just factor out one level of lambda as a private member function, which also matches what we do for restore.